src/Stormpath/Util/UserAgentBuilder.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+
+namespace Stormpath\Util;
+
+use Stormpath\Exceptions\UserAgentException;
+
+class UserAgentBuilder {
+
+    /**
+     * @var string
+     */
+    protected $sdkName = 'stormpath-sdk-php';
+
+    /**
+     * @var string
+     */
+    protected $sdkVersion = Version::SDK_VERSION;
+
+    /**
+     * @var bool
+     */
+    protected $hasIntegration = false;
+
+    /**
+     * @var
+     */
+    protected $phpVersion;
+
+    /**
+     * @var
+     */
+    protected $osVersion;
+
+    /**
+     * @var
+     */
+    protected $osName;
+
+    /**
+     * @codeCoverageIgnore
+     * @param IntegrationUserAgent $userAgent
+     * @return $this
+     */
+    // Todo: Allow setting integration information to the UserAgent
+    // @codeCoverageIgnoreStart
+    public function setIntegration(IntegrationUserAgent $userAgent)
+    {
+
+    }
+    // @codeCoverageIgnoreEnd
+
+    /**
+     * Set the SDK Name
+     * @param string $sdkName
+     * @return $this
+     */
+    public function setSdkName($sdkName = 'stormpath-sdk-php')
+    {
+        $this->sdkName = $sdkName;
+        return $this;
+    }
+
+    /**
+     * Set the SDK Version
+     * @param string $sdkVersion
+     * @return $this
+     */
+    public function setSdkVersion($sdkVersion = Version::SDK_VERSION)
```

Comment:
I think we don't need setters for sdkVersion and sdkName because these will always be constants.